speaker/main.go

```diff
@@ -264,6 +217,12 @@ func (c *controller) SetBalancer(l gokitlog.Logger, name string, svc *v1.Service
 	l.Log("event", "startUpdate", "msg", "start of service update")
 	defer l.Log("event", "endUpdate", "msg", "end of service update")
 
+	if svc.ObjectMeta.Namespace == c.namespace && svc.ObjectMeta.Name == c.service {
+		if p, ok := c.protocols[config.Layer2]; ok {
+			p.(*layer2Controller).SetSpeakers(l, eps)
```

Comment:
I don't think it makes sense calling this here. This is registered as a serviceChange callback, but we have NodeChange callbacks too, for example. Wouldn't code be way simpler if we call it there?

Reply:
Yes this part I was not sure, I'll wire up some SetSpeaker callback

Reply:
Cool! :)

Reply:
done